# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../pam_ldap/pam_ldap-178-autotools.patch
# Copyright (C) 2004 - 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -urN pam_ldap-180.orig/configure.in pam_ldap-180/configure.in
--- pam_ldap-180.orig/configure.in	2005-08-18 00:35:13.000000000 +0200
+++ pam_ldap-180/configure.in	2005-09-06 08:41:23.192126000 +0200
@@ -13,8 +13,17 @@
 AC_ARG_ENABLE(ssl, [  --disable-ssl           disable SSL/TSL support])
 AC_ARG_WITH(ldap-lib, [  --with-ldap-lib=type    select ldap library [auto|netscape5|netscape4|netscape3|umich|openldap]])
 AC_ARG_WITH(ldap-dir, [  --with-ldap-dir=DIR     base directory of ldap SDK])
-AC_ARG_WITH(ldap-conf-file, [  --with-ldap-conf-file     path to LDAP configuration file], [AC_DEFINE_UNQUOTED(PAM_LDAP_PATH_CONF, "$with_ldap_conf_file")])
-AC_ARG_WITH(ldap-secret-file, [  --with-ldap-secret-file   path to LDAP root secret file], [AC_DEFINE_UNQUOTED(PAM_LDAP_PATH_ROOTPASSWD, "$with_ldap_secret_file")])
+AC_ARG_WITH(ldap-conf-file, [  --with-ldap-conf-file     path to LDAP configuration file],
+	    [ PAM_LDAP_PATH_CONF=$with_ldap_conf_file ],
+	    [ PAM_LDAP_PATH_CONF="/etc/ldap.conf" ])
+AC_ARG_WITH(ldap-secret-file, [  --with-ldap-secret-file   path to LDAP root secret file],
+	    [ PAM_LDAP_PATH_ROOTPASSWD="$with_ldap_secret_file" ],
+	    [ PAM_LDAP_PATH_ROOTPASSWD="/etc/ldap.secret" ])
+
+AC_DEFINE_UNQUOTED(PAM_LDAP_PATH_CONF, "$PAM_LDAP_PATH_CONF")
+AC_DEFINE_UNQUOTED(PAM_LDAP_PATH_ROOTPASSWD, "$PAM_LDAP_PATH_ROOTPASSWD")
+AC_SUBST(PAM_LDAP_PATH_CONF)
+AC_SUBST(PAM_LDAP_PATH_ROOTPASSWD)
 
 if test "$ac_cv_prog_gcc" = "yes"; then CFLAGS="$CFLAGS -Wall -fPIC"; fi
 
diff -urN pam_ldap-180.orig/Makefile.am pam_ldap-180/Makefile.am
--- pam_ldap-180.orig/Makefile.am	2005-08-18 00:35:13.000000000 +0200
+++ pam_ldap-180/Makefile.am	2005-09-06 08:43:04.938484750 +0200
@@ -1,7 +1,13 @@
-noinst_PROGRAMS = pam_ldap.so
+noinst_PROGRAMS = pam_ldap.so chfn chsh
 EXTRA_DIST = COPYING.LIB CVSVersionInfo.txt ChangeLog README \
 	     ldap.conf pam.conf pam_ldap.spec pam.d
 
+PAM_LDAP_PATH_CONF = @PAM_LDAP_PATH_CONF@
+PAM_LDAP_PATH_ROOTPASSWD = @PAM_LDAP_PATH_ROOTPASSWD@
+
+chfn_SOURCES = chfn.in
+chsh_SOURCES = chsh.in
+
 pam_ldap_so_SOURCES = pam_ldap.c pam_ldap.h md5.c md5.h
 pam_ldap_so_LDFLAGS = @pam_ldap_so_LDFLAGS@
 
@@ -35,11 +41,20 @@
 
 install-data-local:
 	@$(NORMAL_INSTALL)
-	@if test ! -f $(DESTDIR)$(sysconfdir)/ldap.conf; then \
-		$(mkinstalldirs) $(DESTDIR)$(sysconfdir); \
-		$(INSTALL_DATA) -o root -g root $(srcdir)/ldap.conf $(DESTDIR)$(sysconfdir)/ldap.conf; \
+	@if test ! -f $(DESTDIR)$(PAM_LDAP_PATH_CONF); then \
+		$(mkinstalldirs) $(DESTDIR)$(dir $(PAM_LDAP_PATH_CONF)); \
+		$(INSTALL_DATA) -o root -g root $(srcdir)/ldap.conf $(DESTDIR)$(PAM_LDAP_PATH_CONF); \
 	fi
 
 uninstall-local:
 	@$(NORMAL_UNINSTALL)
 
+
+chfn$(EXEEXT):	chfn.in
+
+chsh$(EXEEXT):	chsh.in
+
+%$(EXEEXT)::	%.in
+	sed -e "s:/etc/ldap.conf:$(PAM_LDAP_PATH_CONF):" \
+	    -e "s:/etc/ldap.secret:$(PAM_LDAP_PATH_ROOTPASSWD):" < $< > $@
+
